### PR TITLE
Add ruby-standardrb checker

### PR DESCRIPTION
**Pull request by julianrubisch**
---
name: Ruby Standardrb
about: Provide a new checker implementation
title: Add `ruby-standardrb` checker
labels: 'component: checkers'
assignees: ''

---

## Checklist
- [X] I have read the [Contributor's guide][].
- [ ] I have documented this checker in the [manual][].
- [X] I have added a test for this checker in [flycheck-test.el][].
- [ ] (*If you have written a test*) I have created a companion PR to add the
      checker tool in [flycheck/docker-tools][].
- [ ] I have mentioned the checker in the [Changelog][].

## Description
Describe the checker you added.

## Additional context
Additional relevant information about the checker and its integration into Flycheck.

[Changelog]: https://github.com/flycheck/flycheck/blob/master/CHANGES.rst
[manual]: https://www.flycheck.org/en/latest/languages.html
[flycheck-test.el]: https://github.com/flycheck/flycheck/blob/master/test/flycheck-test.el
[flycheck/docker-tools]: https://github.com/flycheck/docker-tools
[Contributor's guide]: https://www.flycheck.org/en/latest/contributor/contributing.html

### CODE REVIEW

**Comment by julianrubisch** [comment]
manual and changelog entries are missing, sry

**Comment by cpitclaudel** [comment]
Sorry for leaving this unanswered for so long.  This is a great PR.  Do you need help getting the tests passing?

**Comment by julianrubisch** [comment]
I‘m afraid so, yes 😕

**Comment by julianrubisch** [comment]
I _think_ I fixed the unit tests, unsure about the rest...

FWIW, `make check specs unit` runs cleanly locally...

File: test/flycheck-test.el [hunk 1]
```diff
@@ -4225,7 +4236,7 @@ Why not:
      '(5 7 error "unexpected token tCONSTANT" :checker ruby-rubylint))))
 
 (flycheck-ert-def-checker-test ruby ruby syntax-error
-  (let ((flycheck-disabled-checkers '(ruby-rubocop ruby-reek ruby-rubylint)))
```
Comment (fmdkdd) [inline test/flycheck-test.el]: Why did you re-enable the checker in the tests?  I think it's the reason that `make integ` tests fail on the CI.

**Comment by cpitclaudel** [comment]
Thanks, I finally looked into this. There's one thing that confuses me. StandardRB is incompatible with rubocop, and it seems to share its configuration and command line syntax and error patterns with rubocop. Wouldn't it be enough to set the rubocop executable to standardrb to use it? Do we need a second checker definition?

**Comment by julianrubisch** [comment]
well, as I understood it (and guys from testdouble confirm it), the StandardRB is nothing than an opinionated wrapper around Rubocop. so if I understand you correctly, you suggest doing `(flycheck-set-checker-executable standardrb)` ... ?

I see why that would be doable, let me just add that
1. standardrb linter plugins are being developed at the moment and I thought flycheck should have one
2. personally I find it more accessible to do `C-c ! s` and switch checkers if I like to.

But I will admit that your argument has a point, and I won't be disappointed if you close this.

**Comment by cpitclaudel** [comment]
> So if I understand you correctly, you suggest doing `(flycheck-set-checker-executable standardrb)` ... ?

That's right.  But your point is valid too.  Please look at what I just pushed. I rebased on master, shared the error patterns, and shared the lint-only option.  Can you give it a try (I don't have a standardrb install at hand)

**Comment by julianrubisch** [comment]
Looking good on my end!

**Comment by cpitclaudel** [comment]
> Looking good on my end!

And working, too? ^^

**Comment by julianrubisch** [comment]
Yes, sorry, should have been more clear on that. Works like a charm!

Clément Pit-Claudel <notifications@github.com> schrieb am Mi. 26. Feb. 2020
um 13:50:

**Comment by julianrubisch** [comment]
is there anything else I can do to advance this PR?

**Comment by cpitclaudel** [comment]
> is there anything else I can do to advance this PR?

A gentle prod was all it took :)  Congrats on your contribution and thanks a lot for your patience.  Here's to hoping for many more PRs, hopefully with slightly faster turnaround from our side ^^

**Comment by julianrubisch** [comment]
I hope I can be of help in the future, too!

**Pull request merged**